File: Insertion at Tail singly linkedlist.cpp
#include <stdlib.h>
#include <iostream>
using namespace std;
struct Node
{
  int data_part;
  struct Node* next_part;
};
struct Node* head_pointer=NULL;
void create_node_at_End(int value_of_data_part)
{
  struct Node* CreateNewNode = (struct Node*)malloc(sizeof(struct Node));
  CreateNewNode->data_part = value_of_data_part;
  if (head_pointer== NULL)
  {
        CreateNewNode->next_part=NULL;
        head_pointer=CreateNewNode;
  }
  else
  {
      struct Node* temp_pointer=head_pointer;
      while(temp_pointer->next_part!=NULL)
      {
          temp_pointer=temp_pointer->next_part;
      }
      temp_pointer->next_part=CreateNewNode;
      CreateNewNode->next_part=NULL;
  }
}
void display()
{
         if(head_pointer==NULL)
         {
             cout<<"empty";
         }
         else
         {
             struct Node* temp_pointer=head_pointer;
             while(temp_pointer!=NULL)
             {
                 cout<<temp_pointer->data_part<<" ";
                 temp_pointer=temp_pointer->next_part;
             }
         }
}
int main()
{
    cout<<"Inserted Element At End: \n";
    create_node_at_End(21);display();cout<<endl;
    create_node_at_End(7);display();cout<<endl;
    create_node_at_End(8);display();cout<<endl;
    create_node_at_End(50);display();cout<<endl<<endl;
}

File: Insertion at Tail singly linkedlist.py
class NewNode:
    def __init__(self, value_of_Data_part):
        self.data_part =value_of_Data_part
        self.next_part = None

class linked_list:
    def __init__(self):
        self.head_point = None

    def  create_node_at_End(self, value_of_Data_part):
        CreateNewNode = NewNode(value_of_Data_part)
        if self.head_point==None:
            CreateNewNode.next_part=None
            self.head_point=CreateNewNode
        else:
            temp=self.head_point
            while temp.next_part != None:
                temp = temp.next_part
            temp.next_part = CreateNewNode
            CreateNewNode.next_part=None

    def display(self):
        if self.head_point==None:
            print("empty")
        else:
            ptr= self.head_point
            while (ptr):
                print(str(ptr.data_part) + " ", end="")
                ptr = ptr.next_part
            print(" ")

ob = linked_list()
print("Inserted Element At End: ")
ob.create_node_at_End(21)
ob.display()
ob.create_node_at_End(7)
ob.display()
ob.create_node_at_End(8)
ob.display()
ob.create_node_at_End(50)
ob.display()
print(" ")
